pgmax/fg/graph.py

```diff
@@ -180,31 +183,66 @@ def _register_factor_group(
                 self._factor_group_to_msgs_starts[factor_group]
                 + factor_num_states_cumsum[vv]
             )
-            self._factor_to_potentials_starts[factor] = (
-                self._factor_group_to_potentials_starts[factor_group]
-                + vv * factor.log_potentials.shape[0]
-            )
-            factor_group_num_configs += factor.log_potentials.shape[0]
+
+            if not isinstance(factor_group, groups.ORFactorGroup):
+                self._factor_to_potentials_starts[factor] = (
+                    self._factor_group_to_potentials_starts[factor_group]
+                    + vv * factor.log_potentials.shape[0]
+                )
+                factor_group_num_configs += factor.log_potentials.shape[0]
 
         self._total_factor_num_states += factor_num_states_cumsum[-1]
-        self._total_factor_num_configs += factor_group_num_configs
         if name is not None:
             self._named_factor_groups[name] = factor_group
 
+        if not isinstance(factor_group, groups.ORFactorGroup):
+            self._total_factor_num_configs += factor_group_num_configs
+
     @cached_property
-    def wiring(self) -> nodes.EnumerationWiring:
+    def enum_wiring(self) -> None or nodes.EnumerationWiring:
         """Function to compile wiring for belief propagation.
 
         If wiring has already beeen compiled, do nothing.
 
         Returns:
             Compiled wiring from individual factors.
+            None if the graph only has OR factors.
         """
-        wirings = [
-            factor.compile_wiring(self._vars_to_starts) for factor in self.factors
+        if np.all([isinstance(factor, nodes.ORFactor) for factor in self.factors]):
+            return None
+
+        enum_wirings = [
+            factor.compile_wiring(self._vars_to_starts)
+            for factor in self.factors
+            if isinstance(factor, nodes.EnumerationFactor)
         ]
-        wiring = fg_utils.concatenate_enumeration_wirings(wirings)
-        return wiring
+        enum_wiring = fg_utils.concatenate_enumeration_wirings(enum_wirings)
+        return enum_wiring
+
+    @cached_property
+    def or_wiring(self) -> None or nodes.ORWiring:
+        """Function to compile OR wiring for belief propagation.
+
+        If wiring has already beeen compiled, do nothing.
+
+        Returns:
+            Compiled wiring from individual factors.
+            None if the graph only has Enumeration factors.
+        """
+        if np.all(
+            [
+                isinstance(factor_group, nodes.EnumerationFactor)
+                for factor_group in self.factor_groups
+            ]
+        ):
+            return None
+
+        # or_wirings = [
+        #     factor.compile_wiring(self._vars_to_starts) for factor in self.factors if isinstance(factor, nodes.ORFactor)
+        # ]
+        # or_wiring = fg_utils.concatenate_or_wirings(or_wirings)
```

Comment:
Similar comment as above. Might need to be careful about how concatenation is done.